demos/tests/run_tests.py

```diff
@@ -143,13 +157,29 @@ def option_to_args(key, value):
                     ' '.join(shlex.quote(str(arg)) for arg in case_args))
                 print(flush=True)
 
+                execution_time = -1
+                device = ""
+                pipeline = []
+                for key, value in test_case.options.items():
+                    match = re.search(r'-d.*', key)
+                    if match and device == "":
+                        device = value
+                    match = re.search(r'-m.*', key)
+                    if match:
+                        model_path = option_to_args(key, value)[1]
+                        model = re.search(r'.*/(models/.*).xml', model_path)
+                        pipeline.append(model.group(1) if model else option_to_args(key, value)[1])
```

Comment:
This is overly complicated. For a model option, the value should be a `ModelArg`. You can get the model name directly from its attributes.